contrib/node/src/python/pants/contrib/node/subsystems/resolvers/npm_resolver.py

```diff
@@ -55,7 +55,7 @@ def resolve_target(self, node_task, target, results_dir, node_paths):
           raise TaskError(
             'Cannot find yarn.lock. Did you forget to put it in target sources?')
         returncode, yarnpkg_command = node_task.execute_yarnpkg(
-          args=[],
+          args=['--ignore-optional'],
```

Comment:
Should this be exposed as an option for both package managers?

Reply:
Done.